src/main/java/edu/hm/hafner/analysis/parser/EclipseXMLParser.java

```diff
@@ -0,0 +1,98 @@
+package edu.hm.hafner.analysis.parser;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.util.function.Function;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import org.apache.commons.io.input.ReaderInputStream;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import edu.hm.hafner.analysis.AbstractParser;
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.ParsingCanceledException;
+import edu.hm.hafner.analysis.ParsingException;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.XmlElementUtil;
+
+/**
+ * Parser for Eclipse Compiler output in XML format.
+ * 
+ * @author Jason Faust
+ */
+public class EclipseXMLParser extends AbstractParser {
+
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public Report parse(final Reader reader, final Function<String, String> preProcessor)
+            throws ParsingCanceledException, ParsingException {
+
+        try (InputStream input = new ReaderInputStream(reader, StandardCharsets.UTF_8)) {
```

Comment:
The encoding already has been provided by the base class method (can be changed by user). Wouldn't it make more sense to use the reader as such? Or is UTF8 fixed?

Reply:
I'm honestly not sure.

The use of the ReaderInputStream is due to DocumentBuilder.parse() not accepting Readers, only InputStreams, and ReaderInputStream requires a character set. But the Reader doesn't expose the character set it's generating characters at, but I'd assume (and it's not in the JavaDoc of Reader) that they are code points, so it would put it at UTF-16?

Hurrah for undocumented behavior. I need to test something with how input above 0x7f and some of the surrogate pairs come though.  I'll get back to you.

Reply:
Well I got my wires completely crossed on this.  The passed in character set is for what the InputStream will produce, not what the Reader is returning.

```
public static void main(String[] args) throws Exception {
    Reader sr = new StringReader("𐌰"); // U+10330
    int c;
    sr.mark(0);
    while ((c = sr.read()) != -1) {
        System.out.printf("0x%02x ", c);
    }
    System.out.println();
    for (Charset set : new Charset[] {
            Charset.forName("UTF-8"),
            Charset.forName("UTF-16"),
            Charset.forName("UTF-32") }) {
        sr.reset();
        System.out.print(set + "\t");
        InputStream in = new ReaderInputStream(sr, set);
        while ((c = in.read()) != -1) {
            System.out.printf("0x%02x ", c);
        }
        System.out.println();
    }
}
```
```
0xd800 0xdf30 
UTF-8	0xf0 0x90 0x8c 0xb0 
UTF-16	0xfe 0xff 0xd8 0x00 0xdf 0x30 
UTF-32	0x00 0x01 0x03 0x30 
```

Anyway, all a mute point because can just use parse around an `org.xml.sax.InputSource`, which can wrap the Reader directly.